### DISCUSSION TITLE
unsuccessful install on macos

### DISCUSSION THREAD

**Discussion by selipot** [Parcels-code/Parcels#1150] [Q&A] [answered]
Hello,

I am unable to run parcels on my mac (architecture x86_64) with macos 12.3. I run into compilation errors as soon as I tried any of the tutorials. I have tried the installation in two ways: installing parcels with conda or following the "Installation for developers". In both cases I get a version of the error message below. Looks like there is an issue with XCode? (otherwise I have been able to install parcels on the HPC of my university). Any suggestion on how to resolve this would be greatly appreciated.

```
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libm.tbd' for architecture x86_64
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by lvankampenhout** [comment]
Hello Shane,

This seems to be an issue with your compiler, please try and upgrade Xcode Command Line Tools:

https://stackoverflow.com/questions/42538171/how-to-update-xcode-command-line-tools#53364628

Does that help?

Leo


On 17 Mar 2022, at 13:50, Shane Elipot ***@***.******@***.***>> wrote:



Hello,

I am unable to run parcels on my mac (architecture x86_64) with macos 12.3. I run into compilation errors as soon as I tried any of the titorials. I have tried the installation in two ways: installing parcels with conda or following the "Installation for developers". In both cases I get a version of the error message below. Looks like there is an issue with XCode? (otherwise I have been able to install parcels on the HPC of my university). Any suggestion on how to resolve this would be greatly appreciated.

ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libm.tbd' for architecture x86_64
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)


—
Reply to this email directly, view it on GitHub<https://github.com/OceanParcels/parcels/discussions/1150>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB4Q7HKQ5FB3ZO244X7V4RDVAMTCBANCNFSM5Q634GIA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by selipot** [comment]
I have XCode 13.3 installed, and installed the additional component from within XCode. 

`softwareupdate --all --install --force` returns nothing and `sudo xcode-select --install` returns that "command line tools are already installed"

So I think XCode is up to date?

**Comment by erikvansebille** [comment, marked as answer]
Have you tried `export CC=gcc` before invoking the python call?

**Reply by selipot** [reply]
Hi @erikvansebille
do you mean before `python setup.py install` within the cloned directory from github?

**Comment by selipot** [comment]
And that did it @erikvansebille ! Thank you so much!

**Reply by erikvansebille** [reply]
No worries! Did you not see this comment in the note of step 5 at https://oceanparcels.org/#installing? Anyways, now it is also documented here for people looking for it in the discussion!

**Reply by selipot** [reply]
Ah! indeed, I missed that comment! Thanks again
